plugins/generator-1.15.2/forge-1.15.2/templates/block.java.ftl

```diff
@@ -277,9 +277,15 @@ public class ${name}Block extends ${JavaModName}Elements.ModElement {
 		}
 		</#if>
 
-		<#if data.lightOpacity == 0>
+		<#if (!data.blockBase?? || data.blockBase == "Leaves") && data.lightOpacity == 0>
 		@Override public boolean propagatesSkylightDown(BlockState state, IBlockReader reader, BlockPos pos) {
-			return true;
+			return <#if data.isWaterloggable>state.getFluidState().isEmpty()<#else>true</#if>;
+		}
+		</#if>
+
+		<#if !data.blockBase?? || data.blockBase == "Leaves" || data.lightOpacity != 0>

```

Comment:
I have just checked and thought ice was semi-opaque for light, but checking now, it seems only leaves change opacity. So I am now confused how this parameter is even used in MC. Any ideas? :D

Reply:
I think it's because ice is non-solid, so the default lightOpacity method checks if it propagates skylight. It doesn't, so the light opacity is set to 1. Not sure why leaves do it like this, but it works.